lnwallet/chainfee/filtermanager.go

```diff
@@ -0,0 +1,285 @@
+	var outboundPeerFilters []SatPerKWeight
+	for _, peerResp := range peerResps {
+		if peerResp.Inbound {
+			continue
+		}
+
+		// The value that bitcoind returns for the "minfeefilter" field
+		// is in fractions of a bitcoin that represents the satoshis
+		// per KvB. We need to convert this fraction to whole satoshis
+		// by multiplying with COIN. Then we need to convert the
+		// sats/KvB to sats/KW.
+		//
+		// Convert the sats/KvB from fractions of a bitcoin to whole
+		// satoshis.
+		filterKVByte := SatPerKVByte(
+			peerResp.MinFeeFilter * btcutil.SatoshiPerBitcoin,
+		)
+
+		if !checkFilterBounds(filterKVByte) {
+			continue
+		}
+
+		// Convert KvB to KW and add it to outboundPeerFilters.
+		outboundPeerFilters = append(
+			outboundPeerFilters, filterKVByte.FeePerKWeight(),
+		)
+	}
+
+	return outboundPeerFilters, nil
+}
+
+func fetchBtcdFilters(client *rpcclient.Client) ([]SatPerKWeight, error) {
+	resp, err := client.GetPeerInfo()
+	if err != nil {
+		return nil, err
+	}
+
+	var outboundPeerFilters []SatPerKWeight
+	for _, peerResp := range resp {
+		// We filter for outbound peers since it is harder for an
+		// attacker to be our outbound peer and therefore harder for
+		// them to manipulate us into broadcasting high-fee or low-fee
+		// transactions.
+		if peerResp.Inbound {
+			continue
+		}
+
+		// The feefilter is already in units of sat/KvB.
+		filter := SatPerKVByte(peerResp.FeeFilter)
+
+		if !checkFilterBounds(filter) {
+			continue
+		}
+
+		outboundPeerFilters = append(
+			outboundPeerFilters, filter.FeePerKWeight(),
+		)
+	}
+
+	return outboundPeerFilters, nil
+}
+
+// updateMedian takes a slice of feefilter values, calculates the median, and
+// adds it to the list of values.
+func (f *filterManager) updateMedian(feeFilters []SatPerKWeight) {
+	// If there are no elements, don't update.
+	numElements := len(feeFilters)
+	if numElements == 0 {
+		return
+	}
+
+	median := med(feeFilters)
+
+	f.valuesMtx.Lock()
+	defer f.valuesMtx.Unlock()
+
+	// Update the list of values and roll off the first element.
+	if len(f.values) < medianWindow {
+		f.values = append(f.values, median)
+	} else {
+		// Take off the value at the front of the slice and add the new
+		// one to the end.
+		f.values = f.values[1:]
+		f.values = append(f.values, median)
+	}
+
+	// Log the new moving median.
+	loggingMedian := med(f.values)
+	log.Debugf("filterManager updated moving median to: %v",
+		loggingMedian.FeePerKVByte())
+}
+
+// checkFilterBounds returns false if the filter is unusable and true if it is.
+func checkFilterBounds(filter SatPerKVByte) bool {
+	// Ignore values of 0 and MaxSatoshi. A value of 0 likely means that
+	// the peer hasn't sent over a feefilter and a value of MaxSatoshi
+	// means the peer is using bitcoind and is in IBD.
+	switch filter {
+	case 0:
+		return false
+
+	case btcutil.MaxSatoshi:
+		return false
+	}
+
+	// TODO: ignore values outside of a certain band (configurable?)
```

Comment:
As in values above our max accepted fee?

Reply:
Yeah. I think we need some sort of ceiling, but I don't know what to base it off of...